server/src/CgiResponse.cpp

```diff
@@ -79,28 +65,51 @@ bool CgiResponse::checkForMandatoryHeaders(string& headerLine)
 	return (false);
 }
 
+string &CgiResponse::getBody()
+{
+	string line = readline(_cgiFD);
+	while (line != "\r")
+		line = readline(_cgiFD);
+	//save the rest in body
+	char buf[1024];
+	int n = 0;
+	while ((n = read(_cgiFD, buf, 1024)) > 0)
+	{
+		buf[n] = '\0';
+		// cout << "buf: " << buf << endl;
+		this->_body.append(buf, n);
+	}
+	return (_body);
+}
+
 void CgiResponse::sendResponse()
 {
-	string headers = this->constructHeader();
-	send(_cgiFD, headers.c_str(), headers.length(), 0);
-	tunnelResponse(_cgiFD, _clientFD);
+	cout << "sending response" << " body" << this->_body << endl;
+	string headers  = "HTTP/1.1 200 OK\r\nContent-Length: " + std::to_string(this->_body.length()) + "\r\n\r\n"; // @Tam please do not send a hardcoded response, what if the cgi crahed or something went wrong?
```

Comment:
is this still needed?

Reply:
fixed with dc9cc883d2cfd93bc5d7bd35739f1eba3d37851c

Reply:
fixed with a84262434613201308be1f5a9f8889051bb2be26